### PR TITLE
Fix for Kotlin version mismatch (Xamarin.Android repo issue 7699)

### PR DESCRIPTION
**Pull request by moljac**
### Does this change any of the generated binding API's?

No.

Already published:

- `com.google.firebase:firebase-perf` - 20.2.0 -> 20.3.0
- `com.google.firebase:firebase-perf-ktx` - 20.2.0 -> 20.3.0
- `com.google.android.libraries.places:places-compat` - 2.1.0 -> 2.2.0

Dependencies:

- `com.google.dagger:dagger` - 2.43.2 -> 2.44.2
- `com.squareup.okhttp3:logging-interceptor` - 4.5.0 -> 4.6.0
- `androidx.appcompat:appcompat` - 1.5.1 -> 1.6.0
- `androidx.exifinterface:exifinterface` - 1.3.4 -> 1.3.5
- `com.google.android.gms:playservices-ads-base` - 20.0.0 -> 21.4.0
- `com.google.android.gms:playservices-analytics-impl` - 17.0.1 -> 18.0.2
- `com.google.android.gms:playservices-measurement-sdk` - 21.1.1 -> 21.2.0
- `com.google.auto.value:autovalue-annotations` - 1.9.0 -> 1.10.1
- `com.google.code.gson:gson` - 2.9.0 -> 2.10.1
- `io.reactivex.rxjava2:rxjava` - 2.2.7 -> 2.2.21
- `org.jetbrains.kotlin:kotlin-stdlib` - 1.7.22 -> 1.8.0
- `org.jetbrains.kotlin:kotlin-stdlib-common` - 1.7.22 -> 1.8.0
- `org.jetbrains.kotlin:kotlin-stdlib-jdk7` - 1.7.22 -> 1.8.0
- `org.jetbrains.kotlin:kotlin-stdlib-jdk8` - 1.7.22 -> 1.8.0
- `com.google.android.gms:playservices-measurement-base` - 21.1.0 -> 16.3.0

### Describe your contribution

Nuget package versions bumped in order to align dependencies and fix issues with kotlin:

https://github.com/xamarin/xamarin-android/issues/7699

related

https://github.com/xamarin/AndroidX/pull/689

### CODE REVIEW

File: config.json [hunk 1]
```diff
@@ -29,1615 +29,1615 @@
...
+        "nugetVersion": "119.0.6.8",
         "nugetId": "Xamarin.Firebase.ML.Vision.Object.Detection.Model",
         "dependencyOnly": false
       },
       {
         "groupId": "com.google.firebase",
         "artifactId": "firebase-perf",
-        "version": "20.2.0",
-        "nugetVersion": "120.2.0",
+        "version": "20.3.0",
```
Comment (jpobst) [inline config.json:1319]: This is not just a patch bump, this is a new Java version.

File: config.json [hunk 2]
```diff
@@ -29,1615 +29,1615 @@
...
+        "nugetVersion": "120.3.0.1",
         "nugetId": "Xamarin.Firebase.Perf",
         "dependencyOnly": false
       },
       {
         "groupId": "com.google.firebase",
         "artifactId": "firebase-perf-ktx",
-        "version": "20.2.0",
-        "nugetVersion": "120.2.0",
+        "version": "20.3.0",
```
Comment (jpobst) [inline config.json:1327]: This is not just a patch bump, this is a new Java version.

File: config.json [hunk 3]
```diff
@@ -29,1615 +29,1615 @@
...
+        "nugetVersion": "2.2.0.1",
         "nugetId": "Xamarin.GoogleAndroid.Libraries.Places",
         "dependencyOnly": false
       },
       {
         "groupId": "com.google.android.libraries.places",
         "artifactId": "places-compat",
-        "version": "2.1.0",
-        "nugetVersion": "2.1.0",
+        "version": "2.2.0",
```
Comment (jpobst) [inline config.json:1599]: This is not just a patch bump, this is a new Java version.

File: config.json [hunk 4]
```diff
@@ -1727,16 +1727,16 @@
...
         "nugetId": "Xamarin.Google.Code.FindBugs.JSR305",
         "dependencyOnly": false,
         "templateSet": "findbugs"
       },
       {
         "groupId": "com.google.dagger",
         "artifactId": "dagger",
-        "version": "2.43.2",
-        "nugetVersion": "2.43.2",
+        "version": "2.44.2",
```
Comment (jpobst) [inline config.json:1738]: This is not just a patch bump, this is a new Java version.

File: config.json [hunk 5]
```diff
@@ -1801,8 +1801,8 @@
       {
         "groupId": "com.squareup.okhttp3",
         "artifactId": "logging-interceptor",
-        "version": "4.5.0",
-        "nugetVersion": "4.5.0",
+        "version": "4.6.0",
```
Comment (jpobst) [inline config.json:1804]: This is not just a patch bump, this is a new Java version.

File: config.json [hunk 6]
```diff
@@ -2541,8 +2533,8 @@
       {
         "groupId": "com.google.android.gms",
         "artifactId": "play-services-measurement-base",
-        "version": "21.1.0",
-        "nugetVersion": "121.1.0",
+        "version": "16.3.0",
```
Comment (jpobst) [inline config.json:2536]: This should be examined, it's a large downgrade.

File: config.json [hunk 7]
```diff
@@ -29,1615 +29,1615 @@
...
+        "nugetVersion": "120.1.3.8",
         "nugetId": "Xamarin.GooglePlayServices.Vision",
         "dependencyOnly": false
       },
       {
         "groupId": "com.google.android.gms",
         "artifactId": "play-services-vision-common",
         "version": "19.1.3",
-        "nugetVersion": "119.1.3.7",
+        "nugetVersion": "119.1.3.10",
```
Comment (jpobst) [inline config.json:616]: Bumped 3 versions?

File: config.json [hunk 8]
```diff
@@ -29,1615 +29,1615 @@
...
+        "nugetVersion": "123.1.1.1",
         "nugetId": "Xamarin.Firebase.Messaging.Ktx",
         "dependencyOnly": false
       },
       {
         "groupId": "com.google.firebase",
         "artifactId": "firebase-ml-common",
         "version": "22.1.2",
-        "nugetVersion": "122.1.2.7",
+        "nugetVersion": "122.1.2.0",
```
[the remark this answers is not in the document]
Comment (moljac) [reply config.json:1200]: typo

File: config.json (hunk 1 above)
Comment (moljac) [reply config.json:1319]: I had to manually bump some (remember order of last updates FB was 1st but we merged it as last).
Will fix right now.

File: config.json (hunk 2 above)
Comment (moljac) [reply config.json:1327]: I had to manually bump some (remember order of last updates FB was 1st but we merged it as last).
Will fix right now.

File: config.json (hunk 3 above)
Comment (moljac) [reply config.json:1599]: I had to manually bump some (remember order of last updates FB was 1st but we merged it as last).
Will fix right now.

File: config.json (hunk 4 above)
Comment (moljac) [reply config.json:1738]: I had to manually bump some (remember order of last updates FB was 1st but we merged it as last).
Will fix right now.

File: config.json (hunk 5 above)
Comment (moljac) [reply config.json:1804]: I had to manually bump some (remember order of last updates FB was 1st but we merged it as last).
Will fix right now.

File: config.json (hunk 6 above)
Comment (moljac) [reply config.json:2536]: Good spot. 
I had to tune some package versions. OK. I will check this one.

File: config.json (hunk 7 above)
Comment (moljac) [reply config.json:616]: typo

File: config.json (hunk 6 above)
Comment (moljac) [reply config.json:2536]: removal of:
```
      {
        "groupId": "com.google.android.gms",
        "artifactId": "play-services-measurement-base",
        "version": "16.3.0",
        "nugetVersion": "116.3.0",
        "nugetId": "Xamarin.GooglePlayServices.Measurement.Base",
        "dependencyOnly": true
      },
```
causes:
```
No matching artifact config found for: 
			com.google.android.gms.play-services-measurement-base:[16.3.0]
to satisfy dependency of: 
			com.google.firebase.firebase-analytics-impl:16.3.0
```

File: config.json (hunk 1 above)
Comment (moljac) [reply config.json:1319]: fixed 120.3.0.1

File: config.json (hunk 2 above)
Comment (moljac) [reply config.json:1327]: fixed 120.3.0.1

File: config.json (hunk 3 above)
Comment (moljac) [reply config.json:1599]: fixed 2.2.0.1

File: config.json (hunk 4 above)
Comment (moljac) [reply config.json:1738]: fixed 2.44.2.1

File: config.json (hunk 5 above)
Comment (moljac) [reply config.json:1804]: fixed 4.6.0.1

**Review by jpobst — APPROVED** [review]

**Pull request merged**